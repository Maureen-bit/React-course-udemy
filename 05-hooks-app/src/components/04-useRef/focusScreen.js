import React, { useRef } from 'react';
import '../02-useEffect/effects.css';

export const FocusScreen = () => {
    /* Se usa para guardar una referencia mutable, se puede cambiar la variable a donde apunta 
    el useRef sin necesidad de disparar un renderizaciÃ³n de React */
    const inputReference = useRef();

    const handleClick = () => {
        inputReference.current.select();
        console.log(inputReference);
    }

  return (
      <>
        <h1> Focus Screen </h1>
        <hr />

        <input 
            className='form-control'
            placeholder='Tu nombre'
            ref={inputReference}
        />

        <button 
            className='btn btn-outline-primary mt-5'
            onClick={handleClick}
        >
            Focus
        </button>
      </>
  )
};
