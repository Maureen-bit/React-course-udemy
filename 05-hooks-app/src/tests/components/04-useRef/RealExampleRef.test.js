import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('tests in  <RealExampleRef />', () => { 
    /* Shallow: me sirve para evaluar el componente de manera aislada sin montar nigun otro componente */

    const wrapper = shallow( < RealExampleRef  />);
    test('should render correctly', () => { 
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
     })

     test('should show  <MultipleCustomHooks />', () => { 
        wrapper.find('.btn-primary').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
      })
 })