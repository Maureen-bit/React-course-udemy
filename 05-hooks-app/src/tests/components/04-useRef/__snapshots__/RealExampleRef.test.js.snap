// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in  <RealExampleRef /> should render correctly 1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary "
    onClick={[Function]}
  >
    show/hide
  </button>
</Fragment>
`;
